tests/components/doorbird/test_config_flow.py

```diff
@@ -140,10 +140,33 @@ def _get_mock_doorbirdapi_side_effects(ready=None, info=None):
         context={"source": config_entries.SOURCE_ZEROCONF},
         data={
             "properties": {"macaddress": "notdoorbirdoui"},
+            "host": "192.168.1.8",
             "name": "Doorstation - abc123._axis-video._tcp.local.",
         },
     )
     assert result["type"] == "abort"
+    assert result["reason"] == "not_doorbird_device"
+
+
+async def test_form_zeroconf_link_local_ignored(hass):
+    """Test we abort when we get a link local address via zeroconf."""
+    await hass.async_add_executor_job(
```

Comment:
Is this needed for a failed discovery?

Reply:
Yes, it comes in from the `hass` mock otherwise we get a bonus 
`tests/testing_config/home-assistant_v2.db`